### PR TITLE
Create Reviews Component

### PR DESCRIPTION
**Pull request by manar-abed**
relats #58

### CODE REVIEW

**Comment by manar-abed** [comment]
image

File: client/src/components/Review/review.jsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  const { Option } = Select;
+  const [reviews, setReviews] = useState([]);
+  const [saved, setSaved] = useState(false);
+  const [filterVal, setFilterVal] = useState('all');
+
+  useEffect(() => {
+    const source = axios.CancelToken.source();
+    try {
+      const getReviews = async () => {
+        const { data: { data } } = await axios.get('/api/user/review', { cancelToken: source.token });
```
Comment (Raghad-Mezied) [inline client/src/components/Review/review.jsx:21]: Destructure the reviews too

File: client/src/components/Review/review.jsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        setReviews(data.reviews);
+      };
+      getReviews();
+    } catch ({ Response: { data: { message: msg } } }) {
+      message.error(msg);
+    }
+    return () => {
+      source.cancel();
+    };
+  }, [reviews]);
```
Comment (Raghad-Mezied) [inline client/src/components/Review/review.jsx:31]: Your useEffect changes the reviews and runs when the reviews change?

File: client/src/components/Review/review.jsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+  const handleSave = async (interviewId) => {
+    try {
+      const { message: savedMsg } = await axios.patch(`/api/user/interview/review/${interviewId}`);
+      if (savedMsg === 'Successfully updated!') {
```
Comment (Raghad-Mezied) [inline client/src/components/Review/review.jsx:36]: Why this condition?

File: client/src/components/Review/review.jsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            <Select defaultValue="all" style={{ width: 120 }} value={filterVal} onChange={(value) => setFilterVal(value)}>
+              <Option value="all">all</Option>
+              <Option value="saved">saved</Option>
+            </Select>
+          </div>
+          <Comment
+            actions={[
+              <Tooltip key="comment-basic-like" title="Save">
+                <button type="button" onClick={() => handleSave(review.id)} className="saved-btn">
+                  {saved ? <StarFilled /> : <StarOutlined />}
```
Comment (Raghad-Mezied) [inline client/src/components/Review/review.jsx:58]: Is the save for all your reviews?

File: client/yarn.lock [hunk 5]
```diff
@@ -8101,9 +8132,15 @@ rollup-plugin-terser@^7.0.0:
     terser "^5.0.0"
 
 rollup@^2.43.1:
+<<<<<<< HEAD
+  version "2.72.1"
+  resolved "https://registry.yarnpkg.com/rollup/-/rollup-2.72.1.tgz#861c94790537b10008f0ca0fbc60e631aabdd045"
+  integrity sha512-NTc5UGy/NWFGpSqF1lFY8z9Adri6uhyMLI6LvPAXdBKoPRFhIIiBUpt+Qg2awixqO3xvzSijjhnb4+QEZwJmxA==
+=======
```
Comment (Raghad-Mezied) [inline client/yarn.lock:8139]: Solve the conflicts

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

File: client/src/components/ProfileTabs/style.css [hunk 6]
```diff
@@ -8,4 +8,12 @@
...
     justify-content: center;
     align-items: center;
+}
+.profile-tabs-holder {
+    width: 100%;
+    display: flex;
+    justify-content: center;
+}
+.ant-tabs {
+    width: 70%;
```
Comment (Raghad-Mezied) [inline client/src/components/ProfileTabs/style.css:18]: For this to not affect other styles, so add a class for your container and add these styles to these classes inside your container class.

File: client/src/components/Review/review.jsx [hunk 7]
```diff
@@ -0,0 +1,81 @@
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
+import {
+  Comment, Avatar, Tooltip, message, Select,
+} from 'antd';
+import {
+  StarOutlined, StarFilled,
+} from '@ant-design/icons';
+import './review.css';
```
Comment (Raghad-Mezied) [inline client/src/components/Review/review.jsx:9]: Name the files index.jsx and style.css as we already inside the Review folder

File: client/src/components/Review/review.jsx [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+    const source = axios.CancelToken.source();
+    try {
+      const getReviews = async () => {
+        const { data: { data: { reviews } } } = await axios.get('/api/user/review', { cancelToken: source.token });
+        setReviews(reviews);
+      };
+      getReviews();
+    } catch ({ Response: { data: { message: msg } } }) {
+      message.error(msg);
+    }
```
Comment (Raghad-Mezied) [inline client/src/components/Review/review.jsx:27]: Use the try and catch inside the getReviews function

File: client/src/components/Review/review.jsx [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+  };
+
+  return (
+    <div className="review-tab">
+      {reviewsArr.length > 0 ? reviewsArr.map((review) => (
+        <>
+          <div className="filter-holder">
+            <Select defaultValue="all" style={{ width: 120 }} value={filterVal} onChange={(value) => setFilterVal(value)}>
+              <Option value="all">all</Option>
+              <Option value="saved">saved</Option>
```
Comment (Raghad-Mezied) [inline client/src/components/Review/review.jsx:50]: Don't we have an option for the unsaved reviews?

File: client/src/components/Review/review.jsx [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+      message.error(msg);
+    }
+  };
+
+  return (
+    <div className="review-tab">
+      {reviewsArr.length > 0 ? reviewsArr.map((review) => (
+        <>
+          <div className="filter-holder">
+            <Select defaultValue="all" style={{ width: 120 }} value={filterVal} onChange={(value) => setFilterVal(value)}>
```
Comment (Raghad-Mezied) [inline client/src/components/Review/review.jsx:48]: Move all inline styles to CSS file

File: client/src/components/Review/review.jsx [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+      message.success(savedMsg);
+      setSaved(!saved);
+    } catch ({ Response: { data: { message: msg } } }) {
+      message.error(msg);
+    }
+  };
+
+  return (
+    <div className="review-tab">
+      {reviewsArr.length > 0 ? reviewsArr.map((review) => (
```
Comment (Raghad-Mezied) [inline client/src/components/Review/review.jsx:45]: Destructure the data you need from the review

File: client/src/components/Review/review.jsx [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+            )}
+            datetime={(
+              <Tooltip>
+                <span>{review.review.created_at}</span>
+              </Tooltip>
+            )}
+          />
+
+        </>
+      )) : (<div className="alert alert-primary"> There is no reviews until now </div>) }
```
Comment (Raghad-Mezied) [inline client/src/components/Review/review.jsx:77]: Check the empty antd component.

File: client/src/components/Review/review.jsx [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+          <Comment
+            actions={[
+              <Tooltip key="comment-basic-like" title="Save">
+                <button type="button" onClick={() => handleSave(review.id)} className="saved-btn">
+                  {saved ? <StarFilled /> : <StarOutlined />}
+                </button>
+              </Tooltip>,
+            ]}
+            author={review.interviewerName}
+            avatar={<Avatar src={review.interviewerImage} alt="Han Solo" />}
```
Comment (Raghad-Mezied) [inline client/src/components/Review/review.jsx:62]: Add more descriptive alt

File: client/src/components/index.js [hunk 14]
```diff
@@ -0,0 +1,7 @@
+// eslint-disable-next-line import/no-named-as-default
```
Comment (Raghad-Mezied) [inline client/src/components/index.js:1]: Fix eslint

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

**Review by abedshamia — APPROVED** [review]

**Review by Mahmoud-Ahmad2 — APPROVED** [review]

**Pull request merged**